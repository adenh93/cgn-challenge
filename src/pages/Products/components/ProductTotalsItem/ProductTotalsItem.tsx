import React, { SFC } from "react";
import { Typography, CircularProgress } from "@material-ui/core";
import useStyles from "./styles";

export interface ProductTotalsItemProps {
  label: string;
  amount: number;
  isWaiting: boolean;
}

const ProductTotalsItem: SFC<ProductTotalsItemProps> = ({
  label,
  amount,
  isWaiting
}) => {
  const { root, small } = useStyles();

  return (
    <Typography className={root} component="h6">
      {label}:{" "}
      {isWaiting ? (
        <CircularProgress size={15} />
      ) : (
        <small className={small}>${amount.toFixed(2)}</small>
      )}
    </Typography>
  );
};

export default ProductTotalsItem;
