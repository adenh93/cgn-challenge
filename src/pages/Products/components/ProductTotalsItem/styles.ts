import { createStyles, Theme, makeStyles } from "@material-ui/core";

const useStyles = makeStyles(({ palette }: Theme) =>
  createStyles({
    root: {
      fontSize: 19
    },
    small: {
      color: palette.text.secondary
    }
  })
);

export default useStyles;
