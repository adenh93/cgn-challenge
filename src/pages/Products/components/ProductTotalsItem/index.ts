import ProductTotalsItem from "./ProductTotalsItem";
export default ProductTotalsItem;
