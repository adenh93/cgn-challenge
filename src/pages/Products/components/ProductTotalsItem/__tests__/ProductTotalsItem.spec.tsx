import * as React from "react";
import { create } from "react-test-renderer";
import ProductTotalsItem, {
  ProductTotalsItemProps
} from "../ProductTotalsItem";
import { CircularProgress } from "@material-ui/core";

const props: ProductTotalsItemProps = {
  label: "Total",
  amount: 300,
  isWaiting: false
};

describe("ProductTotals Component", () => {
  test("it matches the snapshot", () => {
    const component = create(<ProductTotalsItem {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("it renders a CircularProgress component if an api call is in progress", () => {
    const { root } = create(
      <ProductTotalsItem {...{ ...props, isWaiting: true }} />
    );
    expect(root.findByType(CircularProgress)).toBeDefined();
  });
});
