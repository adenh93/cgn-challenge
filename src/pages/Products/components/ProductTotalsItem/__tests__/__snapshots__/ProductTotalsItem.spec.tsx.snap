// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTotals Component it matches the snapshot 1`] = `
<h6
  className="MuiTypography-root makeStyles-root-1 MuiTypography-body1"
>
  Total
  :
   
  <small
    className="makeStyles-small-2"
  >
    $
    300.00
  </small>
</h6>
`;
